platform/android/export/export_plugin.cpp

```diff
@@ -3283,12 +3270,16 @@ Error EditorExportPlatformAndroid::export_project_helper(const Ref<EditorExportP
 			cmdline.push_back("clean");
 		}
 
+		String edition = "Standard";
+#ifdef MODULE_MONO_ENABLED
+		edition = "Mono";
+#endif
```

Comment:
This is not ideal because it means every gradle build using the Mono build of the Godot editor will use the `mono` edition, but this should only happen if there's a C# project. The C# `ExportPlugin` sets a feature but I don't think I can access that from here. Alternatively, I could check if the `.csproj` file exists like we do in [`ProjectSettings`](https://github.com/godotengine/godot/blob/d282e4f0e6b6ebcf3bd6e05cd62f2a8fe1f9a238/core/config/project_settings.cpp#L985-L999).

Personally I'm not a big fan of how much C# support is leaking into other areas of the engine.

Reply:
You can use [`EditorExportPlugin#_get_export_features(...)`](https://docs.godotengine.org/en/stable/classes/class_editorexportplugin.html#class-editorexportplugin-private-method-get-export-features) to retrieve the set of features from the C# `ExportPlugin`, check  if there's a C# project, and update the build command accordingly.

Something like:

```
static bool _is_dotnet_project() {
    Vector<Ref<EditorExportPlugin>> export_plugins = EditorExport::get_singleton()->get_export_plugins();
    for (int i = 0; i < export_plugins.size(); i++) {
	if (export_plugins[i]->supports_platform(Ref<EditorExportPlatform>(this))) {
		PackedStringArray export_features = export_plugins[i]->_get_export_features(Ref<EditorExportPlatform>(this), p_debug);
                // Check for 'dotnet' feature
}
```

Reply:
Thanks, I did something similar. But I'm not calling `supports_platform`, to match `EditorExportPlatform`:

https://github.com/godotengine/godot/blob/bdc0316217940a8ccc80ce536547d42e6477adf4/editor/export/editor_export_platform.cpp#L1374-L1376